src/app/modules/todo/pages/list-todo-page/list-todo-page.component.ts

```diff
@@ -90,34 +99,50 @@ export class ListTodoPageComponent implements OnInit, OnDestroy {
       .pipe(take(1))
       .subscribe((todo: Todo) => {
         if (todo){
-          this.todoService.updateTodo(todo);
-          const alertText = $localize`:@@alert-update-success:You have successfully updated`;
-          this.showAlert(alertText);
+          this.todoService.updateTodo(todo)
+            .then(res => {
+              const alertText = $localize`:@@alert-update-success:You have successfully updated`;
+              this.showAlert(alertText);
+            })
+            .catch(err => {
+              this.showAlert(err.message);
+              this.authService.logout();
```

Comment:
why have to logout ? if any different http errors than 401, like 500, you will logout user ?